vscode-trace-extension/src/json-editor/json-editor.ts

```diff
@@ -0,0 +1,451 @@
+
+            // Create and write config to temp file
+            const defaults = this.schemaService.extractSchemaDefaults(this.schema);
+            await this.fileService.loadJSONConfigFile(this.tempFileUri, defaults);
+
+            // Open the document
+            const document = await vscode.workspace.openTextDocument(this.tempFileUri);
+            // JSONC enables comments
+            vscode.languages.setTextDocumentLanguage(document, 'jsonc');
+
+            this.trackedEditor = await vscode.window.showTextDocument(document, {
+                viewColumn: vscode.ViewColumn.Beside
+            });
+            await this.schemaService.setOpenFileSchema(this.tempFileUri, this.schema);
+
+            // Return a promise that resolves when the editor is closed
+            return new Promise<CustomizationSubmission>((resolve, reject) => {
+                this.closeSubscription = vscode.window.onDidChangeVisibleTextEditors(async editors => {
+                    if (this.trackedEditor && !editors.some(e => e.document.uri.fsPath === this.tempFilePath)) {
+                        if (this.closeSubscription) {
+                            this.closeSubscription.dispose();
+                        }
+
+                        try {
+                            const result = await this.handleDocumentClose();
+                            if (result) {
+                                resolve(result);
+                            } else {
+                                reject(new Error('Configuration was not submitted'));
+                            }
+                        } catch (error) {
+                            reject(error);
+                        } finally {
+                            this.trackedEditor = undefined;
+                            this.isEditing = false;
+                            this.fileService.cleanupTempFile(this.tempFilePath);
+                            this.tempFilePath = '';
+                        }
+                    }
+                });
+            });
+        } catch (error) {
+            vscode.window.showErrorMessage(error instanceof Error ? error.message : String(error));
+            return Promise.reject(error instanceof Error ? error : new Error(String(error)));
+        }
+    }
+
+    /**
+     * Validates current configuration and closes the editor if valid
+     */
+    public async closeIfValid(): Promise<void> {
+        if (!this.isEditing || !this.trackedEditor) {
+            throw new Error('No active editor to close');
+        }
+
+        const validation = await this.fileService.validateJsonFile(this.tempFileUri, this.selectedConfig.schema);
+
+        if (!validation.isValid && validation.errors) {
+            await this.displayValidationErrorDialogue(
+                `JSON Error - your config was not submitted because it contained validation errors.`,
+                validation.errors
+            );
+            return;
```

Comment:
If you close the editor hoping to submit it by answering Yes, but the config contains errors, you get this error message but since the editor was closed, you can't fix the mistake, you have to start over.

Reply:
I don't think you can stop a user from closing a file.  The way we want to ask the user "Hey, are you sure you want to close this??" and force the file to stay open isn't possible.  

What I did was open the "problems" tab that displays all the errors when the user opens the config file.  That way, they can see the errors more clearly while editing and before closing.

Let me know what you think

Reply:
I think it's ok, it can be blamed on the user ;). It could be improved (maybe having a way to later resume editing of the bad configuration) but for now it's better if the user tries to Submit instead of relying on close & yes/no prompt.